### DISCUSSION TITLE
Mail notify in case of error

### DISCUSSION THREAD

**Discussion by kalleick** [dewenni/ESP_Buderus_KM271#31] [Ideas]
It would be nice if the software could be expanded to include a mail function. 
You could specify smtp server, user and password and the software would then be able to send a mail when a oil heating error occurs

**Comment by dewenni** [comment]
Thanks for that idea!
In general that would be possible.

I have implemented in combination with node-red an notification to Telegram and Pushover if a new Alarm message is arrived.
If you have also an additional tool, you can react an the incoming MQTT alarm message and send an email or any other message by your own.

But for sure, it could be also implemented directly in the esp.

Are there any other user who would be I interested in this?

I also thought about to implement a direct influxDB connection to the code.

**Reply by djborizza** [reply]
Hi,

I'm also interested in a notification. Currently, I'm facing many errors in winter and I want to be notified actively via mail, push service etc.

**Reply by dewenni** [reply]
I have startet to do some tests with a Email implementation.
Let's see where this leads to :-)

**Reply by dewenni** [reply]
@kalleick , @djborizza 
what kind of messages would you like to receive?
I am lucky to have no problems or errors with my Logamatic, therefore I'm not familiar with the error messages and error flags.

I could monitor the 4 Alarm messages and send an email if this messages change?
This should include at least most the single error Flags from Heating circuit/Boiler/Burner I think.

this are the possible alarms:

- "failure outdoor sensor",
- "failure hc1-flow sensor",
- "failure hc1-flow sensor",
- "failure warm water sensor",
- "Warm Water remains col",
- "Fault thermal disinfection",
- "Fault hc1-remotecontrol",
- "Fault hc1-remotecontrol",
- "no communication to hc1-remote control",
- "no communication to hc1-remote control",
- "Fault Burner 1",
- "no connection with boiler 1",
- "internal Error Nr. 1",
- "internal Error Nr. 2",
- "internal Error Nr. 3",
- "internal Error Nr. 4",
- "failure Boiler flow sensor",
- "failure Boiler auxiliary sensor",
- "Boiler remains cold",
- "Fault Burner",
- "Fault safety chain",
- "External fault boiler",
- "failure Exhaust gas sensor",
- "Exhaust limit exceeded",
- "failure Return flow sensor",

**Reply by kalleick** [reply]
Thank you! Yes, if these are the possible faults, then it would be perfect if they could be sent by email after they occur.
The errors don't go away, only the time changes, so that would have to be recognized.

![image](https://github.com/dewenni/ESP_Buderus_KM271/assets/95974901/9e2e9a4f-c280-40fc-ac7c-f96d27d80807)

**Comment by djborizza** [comment]
@dewenni usually I got problems with the burner. Most times the burner gets in error in the mornings and the whole house is cold.
For me I need some kind of notification via mail or some gpio output to add to a shelly or maybe even alexa notification if this is possible.

I can't change the behavior of the burner so I need to stick to the buderus. But your project here will give me more flexibility to go ob with it.

**Reply by kalleick** [reply]
I had that too, the flame tube on the burner was broken and I ended up having to get a new burner. I first helped myself by sending the data to FHEM via MQTT with a fault detection on the log file, which then sent an email. It would have been nicer if the email function had been available in the firmware. @dewenni thanks for your job working on it. 
The problems occurred sporadically and I always went into the basement and reset the burner.
Nevertheless, the cause of the problem has to be remedied. I then got a new burner for around €900.

Here is the bash script from RPI


#!/bin/bash
start_time="$(date --date='2 hours ago' +'%Y-%m-%d_%H')"
end_time="$(date +'%Y-%m-%d_%H')"
jahr=$(date +"%Y")
log_data="$(grep -E "MQTT2_ESP_Buderus_KM271 .*Fehlerspeicher1: Stoerung .*\(>> [[:digit:]]{2}:[[:digit:]]{2} -0 Tage\)" /opt/fhem/log/MQTT2_ESP_Buderus_KM271-${jahr}.log | awk -v start=$start_time -v end=$end_time '{ if ($1"_"$2 >= start && $1"_"$2 <= end) print }')"
if [ ! -z "$log_data" ]; then
    echo "$log_data" | sendEmail -f 'from@xy,de' -t 'empfaenger@xy.de' -u Buderus Stoerung  -m "bitte Heizung prüfen $log_data" -s smtp.xx.xx:25 -xu 'mailuser' -xp 'password' -o tls=auto -o message-charset=UTF8

else
    echo "Nix zu tun"

fi

**Reply by the78mole** [reply]
Hi, I'm the developer of the km271-wifi, dewenni uses. I developed the hardware because I had exactly the same problem. Sometimes when I want to take a shower, the water was cold because of a burner error. 
I disassembled the whole burner unit more than once, cleaned it, put another pump and another nozzle in, removed the reverse protection oil valve,... but it eliminated the problem only for a few days or weeks. I had a technician here for three times, but they all could not find the root cause.
In the end (after about 5 years), I found out, that when I attach the cover to the burner unit, the hoses got stretched and the oil filter (that also converts a single hose to a two hose system) was pulled a bit, so the angle was not exactly vertical and it sucked some air into the oil flow. Now I run the burner without the cover for over a year without problems.
I used Pushover (KM271 -> ESPhome -> Home Assistant -> Pushover) to notify me about errors and I never had to take a cold shower again :-)

**Comment by dewenni** [comment]
I have already a prototype with Email and Telegram Client running, but I have to do some further tests.
Maybe I can upload a Testversion here and you can do also some tests.

**Reply by djborizza** [reply]
Hi, I can do test but I don't have any Hardware yet. I'm on the wait list I guess.

**Reply by the78mole** [reply]
Hardware will arrive arround end of October... It seems that assembly will start, as soon as all components arrived.

**Comment by dewenni** [comment]
Sorry for the delay, but I am still struggling with the memory consumption of the ESP if I add connections with https.
In general I have a prototype with Email, Telegram and Pushover running, but If I activate Email or Telegram with a secured connection, the Heap of the ESP is running out and the the software is not really stable.
I still have to experiment a little to get it stable.

**Reply by dewenni** [reply]
Due to the memory requirement for an SSL connection when sending emails, I decided against implementing it for the time being.
Instead, I have implemented Pushover. I think this is a good alternative to email.
Included in the current release candidate #80

**Reply by dewenni** [reply]
check out the pushover function in release 4.0.0  #82
